projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts

```diff
@@ -0,0 +1,174 @@
+   */
+  page = 1;
+
+  /**
+   * Define the number of records per page
+   */
+  size = 10;
+
+  /**
+   * Search query
+   */
+  q = '';
+
+  /**
+   * Types of resources available
+   */
+  types: {
+    key: string,
+    label: string,
+    component?: Component,
+    total?: number,
+    canAdd?: any,
+    canUpdate?: any,
+    canDelete?: any,
+    canRead?: any,
+    aggregations?: any
+  }[] = [{ key: 'documents', label: 'Documents' }];
+
+  /**
+   * Constructor
+   * @param dialogService - Modal component
+   * @param recordService - Service for managing records
+   * @param toastService - Toast message
+   * @param route - Angular current route
+   * @param router - Angular router
+   */
+  constructor(
+    private route: ActivatedRoute,
+    protected router: Router
+  ) { }
+
+  /**
+   * Component initialisation.
+   */
+  ngOnInit() {
+    const queryParams = this.route.snapshot.queryParams;
+    for (const key in queryParams) {
+      if (['q', 'page', 'size'].includes(key)) {
+        this[key] = queryParams[key];
+      } else {
+        if (Array.isArray(queryParams[key]) === false) {
+          this.aggFilters.push({ key, values: [queryParams[key]] });
+        } else {
+          this.aggFilters.push({ key, values: queryParams[key] });
+        }
+      }
+    }
+
+    // explicitly cast page and size to number
+    this.page = +this.page;
+    this.size = +this.size;
+
+    const data = this.route.snapshot.data;
+    if (data.types) {
+      this.types = data.types;
+    }
+
+    if (typeof data.showSearchInput !== 'undefined') {
+      this.showSearchInput = data.showSearchInput;
+    }
+
+    if (typeof data.adminMode !== 'undefined') {
+      this.adminMode = data.adminMode;
+    }
+
+    if (data.detailUrl) {
+      this.detailUrl = data.detailUrl;
+    }
+
+    const params = this.route.snapshot.params;
+
+    if (!params.type) {
+      throw new Error('Parameter \"type\" not found.');
+    }
+    this.currentType = params.type;
+  }
+
+  /**
+   * Update URL accordingly to parameters given.
+   * @param parameters Parameters to put in url or query string
+   */
+  updateUrl(parameters: any) {
+    const queryParams = {
+      q: parameters.q,
+      page: parameters.page,
+      size: parameters.size
+    };
+
+    const filters = {};
+    for (const filter of parameters.aggFilters) {
+      filters[filter.key] = filter.values;
+    }
+
+    Object.keys(filters).map((key) => {
+      if (filters[key].length > 0) {
+        queryParams[key] = filters[key];
+      }
+    });
+
+    this.router.navigate([this.getCurrentUrl(parameters.currentType)], { queryParams });
+  }
+
+  /**
+   * Return current URL after removing query parameters and updating resource type.
+   *
+   * @param type Resource type
+   * @returns Updated url without query string
+   */
+  private getCurrentUrl(type: string): string {
+    return this.router.url.split('?')[0].replace(/(^.*\/)(.*)$/, `$1${type}`);
+  }
```

Comment:
You can use Angular to parse the URL: https://angular.io/api/router/Router#parseUrl

Reply:
done in other PR